compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1696,7 +1696,7 @@ object SymDenotations {
             c.ensureCompleted()
       end completeChildrenIn
 
-      if is(Sealed) || isAllOf(JavaEnumTrait) then
+      if is(Sealed) || isAllOf(JavaEnumTrait) && isClass then
```

Comment:
Hum. How can a `JavaEnumTrait` *not* be a class?

Reply:
any java enum case will have these flags, because JavaEnumTrait is poorly named (its actually `JavaDefined | Enum`)

Reply:
enum cases also have the `JavaDefined` and `Enum` flags

Reply:
maybe I can change it to `butNot = Case`, but I didn't check, also should I rename the flags?

Reply:
I think if you rename the flag that would be clearer, indeed.

Reply:
see https://github.com/lampepfl/dotty/pull/19074 where I integrate these changes